src/mpid/ch4/src/ch4r_recv.h

```diff
@@ -17,25 +17,31 @@
 static inline int MPIDIG_reply_ssend(MPIR_Request * rreq)
 {
     int mpi_errno = MPI_SUCCESS, c;
+    MPIR_Context_id_t context_id = 0;
     MPIDIG_ssend_ack_msg_t ack_msg;
+    MPIR_Comm *comm;
+    MPIDI_av_entry_t *av;
+
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_REPLY_SSEND);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_REPLY_SSEND);
     MPIR_cc_incr(rreq->cc_ptr, &c);
     ack_msg.sreq_ptr = MPIDIG_REQUEST(rreq, req->rreq.peer_req_ptr);
+    context_id = MPIDIG_REQUEST(rreq, context_id);
+    comm = MPIDIG_context_id_to_comm(context_id);
+    av = MPIDIU_comm_rank_to_av(comm, MPIDIG_REQUEST(rreq, rank));
 
 #ifndef MPIDI_CH4_DIRECT_NETMOD
     if (MPIDI_REQUEST(rreq, is_local))
         mpi_errno =
-            MPIDI_SHM_am_isend_reply(MPIDIG_REQUEST(rreq, context_id),
+            MPIDI_SHM_am_isend_reply(context_id,
```

Comment:
This change seems to be inconsistently applied across the NM and SHM am APIs in this commit. Is that intentional? The commit message only refers to AV translation. Perhaps this slipped in and is later modified again?

Reply:
This should not be changed at this stage. It should be changed in the later part where SHM related code is used. I will update.